### PR TITLE
Render Cell Tests

### PR DESCRIPTION
**Pull request by stevejpurves**
DONE:
 - readOnly integration tests have been moved to e2e with their own jest config
 - unit style tests added and passing, with their own jest config that transpiles thebe and dependencies
 - updated ci to run both sets

### CODE REVIEW

**Comment by minrk** [comment]
Awesome! Thanks for adding this. Looks like the only failure is the linter: can you run `npm run fmt` and commit the result?

**Comment by stevejpurves** [comment]
@minrk linted. I had the pre-commit hook in place, no idea why it didn't run.

**Comment by stevejpurves** [comment]
@minrk did you see my comment / question on https://github.com/executablebooks/thebe/issues/228??

**Comment by stevejpurves** [comment]
@minrk i've moved this forward. We can either put this up for review/merge as is? 
putting the jupyter server work (which i'll do next) in a separate PR, or wait.

File: package.json [hunk 1]
```diff
@@ -77,11 +78,16 @@
...
-    "puppeteer": "^5.4.0",
+    "puppeteer": "^5.4.1",
     "style-loader": "^2.0.0",
     "url-loader": "^4.1.1",
     "webpack": "^4.0",
     "webpack-bundle-analyzer": "^3.9.0",
     "webpack-cli": "^4.1.0"
+  },
+  "dependencies": {
+    "identity-obj-proxy": "^3.0.0",
```
Comment (minrk) [inline package.json:89]: I don't think there should be any dependencies added here. Are these only test dependencies?

File: package.json [hunk 2]
```diff
@@ -11,9 +11,10 @@
     "fmt": "prettier --trailing-comma=es5 --write *.js src/*.js examples/*.js test/*.js",
     "serve": "http-server -c-1 -a 127.0.0.1 -o development.html",
     "develop": "concurrently \"npm run build:watch\" \"npm run serve\"",
-    "test": "karma start --single-run --browsers ChromeHeadless karma.conf.js",
-    "test:jest": "jest test.js",
-    "test:watch": "karma start --auto-watch --browsers ChromeHeadless karma.conf.js"
+    "test": "jest",
```
Comment (minrk) [inline package.json:14]: Does this mean karma and mocha can be removed as dependencies?

**Review by minrk — COMMENTED** [review]
This is really nice! I only had a question about the dependencies, but the test changes look great.

**Comment by Miniland1333** [comment]
@stevejpurves, since #297 has been merged in to master would you be able to update from master to resolve these conflicts? And yes, I think you/we should remove the Karma and Mocha dependencies to clean that up.

File: package.json (hunk 2 above)
Comment (stevejpurves) [reply package.json:14]: yes! i need to clean up good point

**Comment by stevejpurves** [comment]
Hi @minrk are we ok to merge this one?

**Comment by moorepants** [comment]
It seems you've addressed @minrk's comments. I'll merge near the end of my work day if there is no more feedback.

**Comment by moorepants** [comment]
> Does this mean karma and mocha can be removed as dependencies?

Maybe this needs to be done in a new PR?

**Pull request merged**